waveformview/src/main/java/space/siy/waveformview/FixedWaveFormView.kt

```diff
@@ -48,34 +50,113 @@ class FixedWaveFormView(
     attr: AttributeSet
   ) : this(context, attr, 0)
 
+  private var barShader =
+    LinearGradient(0f, 0f, 0f, 700f, Color.RED, Color.GRAY, Shader.TileMode.CLAMP)
+  private val blockPaint: Paint
+
+  /**
+   * Width each block
+   */
+   private val blockWidth: Float
+
+  /**
+   * Radius of top dome
+   */
+  private val domeRadius: Float
+
+  /**
+   * Flag to enable/disable top dome drawing
+   */
+  private val domeDrawEnabled: Boolean
+
+  /**
+   * Flag to enable/disable seek on touch/drag
+   */
+  private val seekEnabled: Boolean
+
+  /**
+   * Gap needs to be between two consecutive bar
+   */
+  private val gapWidth: Float
+
+  /**
+   * Scale of top blocks
+   */
+  private val topBlockScale: Float
+
+  /**
+   * Scale of bottom blocks
+   */
+  private val bottomBlockScale: Float
+
   /**
-   * Used to retrieve the values defined in the layout XML
+   * Color used in played blocks
    */
-  private val lp =
-    context.obtainStyledAttributes(attr, R.styleable.FixedWaveFormView, defStyleAttr, 0)
+  private val blockColorPlayed: Int
+
+  /**
+   * Color used in blocks default
+   */
+  private val blockColor: Int
+
+  init {
+    val lp =
+      context.obtainStyledAttributes(attr, R.styleable.FixedWaveFormView, defStyleAttr, 0)
+    blockWidth = lp.getDimension(R.styleable.FixedWaveFormView_blockWidth, 5f)
+    gapWidth = lp.getDimension(R.styleable.FixedWaveFormView_gapWidth, 2f)
+    domeDrawEnabled = lp.getBoolean(R.styleable.FixedWaveFormView_domeDrawEnabled, false)
+    seekEnabled = lp.getBoolean(R.styleable.FixedWaveFormView_seekEnabled, false)
+    topBlockScale = lp.getFloat(R.styleable.FixedWaveFormView_topBlockScale, 1f)
+    bottomBlockScale = lp.getFloat(R.styleable.FixedWaveFormView_bottomBlockScale, 0f)
+    blockColor = lp.getColor(R.styleable.FixedWaveFormView_blockColor, Color.WHITE)
+    blockColorPlayed = lp.getColor(R.styleable.FixedWaveFormView_blockColorPlayed, Color.RED)
+    blockPaint = Paint()
+    domeRadius = blockWidth / 2
+    lp.recycle()
+  }
+
+  private var upperWaveBars: Array<RectF>? = null
+  private var bottomWaveBars: Array<RectF>? = null
+
+  // api
+  var duration: Long = 1L // We use this in some division, so don't make it zero
```

Comment:
why 1L and not -1L since -1 should be an invalid value?

Reply:
This is used in some division, which updates the ui. Resulting in negative value can create a bunch of ui glitches and break some assumptions in those part.